x-pack/test/functional/services/observability/alerts/pagination.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+const ALERTS_ROWS_PER_PAGE_SELECTOR = 'tablePaginationPopoverButton';
+const ALERTS_PAGINATION_BUTTON_PREVIOUS = 'pagination-button-previous';
+const ALERTS_PAGINATION_BUTTON_NEXT = 'pagination-button-next';
+const ALERTS_PAGINATION_TEN_ROWS = 'tablePagination-10-rows';
+const ALERTS_PAGINATION_TWENTY_FIVE_ROWS = 'tablePagination-25-rows';
+const ALERTS_PAGINATION_FIFTY_ROWS = 'tablePagination-50-rows';
+const ALERTS_PAGINATION_BUTTON_ONE = 'pagination-button-0';
+const ALERTS_PAGINATION_BUTTON_TWO = 'pagination-button-1';
+
+export function ObservabilityAlertsPaginationProvider({ getService }: FtrProviderContext) {
+  const testSubjects = getService('testSubjects');
+
+  const getPageSizeSelector = async () => {
```

Comment:
Nit: These properties seem to be sorted by which page element they are using, but it's still a flat list. Would it be helpful to group related properties in an object? 

I'm thinking something like:
```
const pageSizeSelector = {
    get: async () => { ...
    getOrFail: async () => { ...
    getOrMissing: async () => { ...
}
```

That would make it faster to scan which properties exist for a particular page element. A middle ground between having many small providers while still helping with some readability.
Not sure what would be done for those that only have function.